### PR TITLE
N°5937 - VSphere Collector - Datastore collect

### PR DESCRIPTION
**Pull request by xtophe38**
Add a new collector to retrieve Datastores from vMWare

### CODE REVIEW

File: src/vSphereDatastoreCollector.class.inc.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			$aDatastoreData = array(
+				'id' => $oDatastore->getReferenceId(),
+				'primary_key' => $oDatastore->getReferenceId(),
+				'name' => $oDatastore->name,
+				'uuid' => $oDatastore->getReferenceId(),
+				'org_id' => $sDefaultOrg,
+				'type' => strtolower($oDatastore->summary->type),
+				'capacity' => (int)($oDatastore->summary->capacity / (1024 * 1024 * 1024)),
+				'status' => 'production',
+				'mounting_point' => $oDatastore->summary->url,
```
Comment (Hipska) [inline src/vSphereDatastoreCollector.class.inc.php:60]: `mounting_point` vs `mountingpoint` in the constructor.

File: src/vSphereDatastoreCollector.class.inc.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		return true;
+	}
+
+	public function Fetch()
+	{
+		if ($this->idx < count($this->aDatastores)) {
+			$aDS = $this->aDatastores[$this->idx++];
+			$aResult = array();
+			foreach ($this->aDatastoreFields as $sAttCode) {
+				$aResult[$sAttCode] = $aDS[$sAttCode];
```
Comment (Hipska) [inline src/vSphereDatastoreCollector.class.inc.php:86]: No check if the key actually exists for the item...

**Review by Hipska — CHANGES_REQUESTED** [review]

File: src/vSphereDatastoreCollector.class.inc.php (hunk 1 above)
Comment (xtophe38) [reply src/vSphereDatastoreCollector.class.inc.php]: Corrected

File: src/vSphereDatastoreCollector.class.inc.php (hunk 2 above)
Comment (xtophe38) [reply src/vSphereDatastoreCollector.class.inc.php]: Check added to $aDS

**Comment by xtophe38** [comment]
Further work will probably be needed to refine Datastore collection.

**Pull request merged**